### PR TITLE
[sitecore-jss][Next.js] Introduce performance metrics for debug logging

### PR DESCRIPTION
**Pull request by illiakovalenko**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated

## Description / Motivation

* Introduced **common** debug namespace
* Added timestamp for base services, nextjs middlewares, page-props-factory

## Testing Details

- [x] Unit Test Added
- [x] Manual Test/Other - verified that timestamp is present in debug log

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by sc-addypathania — APPROVED** [review]
Looks great 💯

File: packages/create-sitecore-jss/src/templates/nextjs/src/lib/middleware/index.ts [hunk 1]
```diff
@@ -19,7 +20,15 @@ export default async function middleware(
 ): Promise<NextResponse> {
   const response = NextResponse.next();
 
-  return (Object.values(plugins) as MiddlewarePlugin[])
+  debug.common('next middleware start');
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/templates/nextjs/src/lib/middleware/index.ts:23]: Most (if not all) middleware has specific debug namespaces with performance metrics included in the logs now. Won't these common middleware messages be extra noise in most cases?

**Review by ambrauer — COMMENTED** [review]
See one comment. Also, wonder if we should retire the `http` namespace (our fallback for all 3 request clients) and use `common` instead?

File: packages/create-sitecore-jss/src/templates/nextjs/src/lib/middleware/index.ts (hunk 1 above)
Comment (illiakovalenko) [reply packages/create-sitecore-jss/src/templates/nextjs/src/lib/middleware/index.ts:23]: The idea was to get the total metric for next middleware, if we will remove that, it requires calculating it

Comment (ambrauer) [reply packages/create-sitecore-jss/src/templates/nextjs/src/lib/middleware/index.ts:23]: Ah, I missed that point. That makes sense.

**Comment by illiakovalenko** [comment]
@ambrauer and I agreed to leave it as it is, better to keep a separation between HTTP logs and common/utils logs that can be added to the sample app

**Pull request merged**